src/ServiceControl.Management.PowerShell/DependencyResolver.cs

```diff
@@ -0,0 +1,106 @@
+﻿#nullable enable
+namespace ServiceControl.Management.PowerShell;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.DependencyModel;
+
+class DependencyResolver
+{
+    readonly string assemblyDirectory;
+    readonly DependencyContext dependencyContext;
+    readonly string?[] runtimes;
+
+    public DependencyResolver(string assemblyPath)
+    {
+        assemblyDirectory = Path.GetDirectoryName(assemblyPath) ?? string.Empty;
+
+        var depsJsonFile = Path.ChangeExtension(assemblyPath, "deps.json");
+        using var fileStream = File.OpenRead(depsJsonFile);
+
+        var reader = new DependencyContextJsonReader();
+        dependencyContext = reader.Read(fileStream);
+
+        var runtimeGraph = DependencyContext.Default?.RuntimeGraph.SingleOrDefault(r => r.Runtime.Equals(RuntimeInformation.RuntimeIdentifier, StringComparison.Ordinal));
+        runtimeGraph ??= DependencyContext.Default?.RuntimeGraph.SingleOrDefault(r => r.Runtime.Equals(DependencyContext.Default.Target.Runtime, StringComparison.Ordinal));
```

Comment:
Add a short comment about what scenario would cause runtimeGraph to be null here.